server/src/main/java/org/elasticsearch/cluster/metadata/MetadataDeleteIndexService.java

```diff
@@ -137,7 +137,11 @@ public static ClusterState deleteIndices(ClusterState currentState, Set<Index> i
             metadataBuilder.remove(indexName);
             if (backingIndices.containsKey(index)) {
                 DataStream parent = metadataBuilder.dataStream(backingIndices.get(index).getName());
-                metadataBuilder.put(parent.removeBackingIndex(index));
+                if (parent.isFailureStoreIndex(indexName)) {
```

Comment:
I can't comment on the line I want because it's not changed, but a couple of lines higher, we check `if (parent.getWriteIndex().equals(im.getIndex()))`. I assume it's intentional that we're not failing when trying to delete the write index of the failure store? If that's correct, maybe add a comment there and perhaps even a test that enforces this behavior?

Reply:
Indeed, I expect the `removeFailureStoreIndex` to handle it the way we want to handle it. No need to duplicate this logic.

Reply:
Hmm doesn't it then make sense to remove the check on line 101 as well? Now it seems (to me) like we behave differently for deleting a backing index vs a failure index, while we actually do the same thing (at least for now until we implement lazy failure store instantiation).

Reply:
The check there is in a way failing faster. We can always remove it if we think it's simpler to have it removed, but I would not do this as part of this PR.

I understand your concern about treating them differently, but since we are still playing with the idea of allowing a failure store to not have a backing index, I think we should not leak the logic of which failure store indices are allowed to be deleted and which ones are not, here. At least not yet.

I do hear your concern, and I think we can address this in https://github.com/elastic/elasticsearch/issues/108722 were deleting the write index comes into play as well.